### PR TITLE
OCD-4660 - Update functionality for SSO flag

### PR DESCRIPTION
**Pull request by tmy1313**
OCD-4660

### CODE REVIEW

File: chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/AuthenticationController.java [hunk 1]
```diff
@@ -59,21 +76,142 @@ public class AuthenticationController {
...
+    @Operation(summary = "Log user out.",
+            description = "Invalidates all of the tokens associated with the user.",
+            security = {
+                    @SecurityRequirement(name = SwaggerSecurityRequirement.API_KEY)
+            }
+        )
+    @RequestMapping(value = "/logout", method = RequestMethod.POST,
+            consumes = MediaType.APPLICATION_JSON_VALUE, produces = "application/json; charset=utf-8")
+    public void logout(@RequestBody CognitoLogoutRequest request) {
+        cognitoAuthenticationManager.invalidateTokensForUser(request.getEmail());
```
Comment (andlar) [inline chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/AuthenticationController.java:139]: Does this need a FF4J check? It looks like most of the other endpoints have one. For the ones that don't, should they as well?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/auth/user/JWTAuthenticatedUser.java [hunk 2]
```diff
@@ -22,6 +23,8 @@
 public class JWTAuthenticatedUser implements UserDetails, Authentication {
     private static final long serialVersionUID = -7558546038256722930L;
 
+    @DeprecatedResponseField(message = "This field is deprecated and will be removed.", removalDate = "2024-11-01")
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/auth/user/JWTAuthenticatedUser.java]: I believe we usually give six months warning on deprecation

Comment (tmy1313) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/auth/user/JWTAuthenticatedUser.java]: This actually doesn't need `@DeprecatedResponseField`.  It is not an object that is publicly available and is only used internally within the API.

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — APPROVED** [review]

File: chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/UserManagementController.java [hunk 3]
```diff
@@ -362,9 +485,9 @@ public DeletedUser deleteUser(@PathVariable("userId") Long userId) throws UserRe
     @PreAuthorize("isAuthenticated()")
     public @ResponseBody UsersResponse getUsers() {
         List<User> users = null;
-        if (AuthUtil.getCurrentUser().getAuthenticationSystem().equals(AuthenticationSystem.COGNITO)) {
+        if (ff4j.check(FeatureList.SSO)) {
```
Comment (kekey1) [inline chpl/chpl-api/src/main/java/gov/healthit/chpl/web/controller/UserManagementController.java:483]: This endpoint is marked as Deprecated/DeprecatedApi but I am not sure that it is. It is being used from the Users -> User Management page in both flag states. If it is truly deprecated then I would expect a different endpoint to be called when the SSO flag is ON.

**Review by kekey1 — CHANGES_REQUESTED** [review]

**Review by kekey1 — APPROVED** [review]

**Review by andlar — APPROVED** [review]

**Pull request merged**